zerver/decorator.py

```diff
@@ -568,17 +568,22 @@ def _wrapped_view_func(request: HttpRequest, *args: object, **kwargs: object) ->
     return cast(ViewFuncT, _wrapped_view_func)  # https://github.com/python/mypy/issues/1927
 
 
-def require_server_admin_api(view_func: ViewFuncT) -> ViewFuncT:
+def require_server_admin_api(
+    view_func: Callable[Concatenate[HttpRequest, ParamT], HttpResponse]
+) -> Callable[Concatenate[HttpRequest, ParamT], HttpResponse]:
     @zulip_login_required
     @wraps(view_func)
     def _wrapped_view_func(
-        request: HttpRequest, user_profile: UserProfile, *args: object, **kwargs: object
+        request: HttpRequest,
+        /,
+        *args: ParamT.args,
+        **kwargs: ParamT.kwargs,
     ) -> HttpResponse:
-        if not user_profile.is_staff:
+        if not request.user.is_staff:
```

Comment:
`request.user` is not the same as `user_profile` in the case of `authenticated_rest_api_view`.

Reply:
Probably we can make this change after the remote server gets refactored out of it. We currently can't mix different concatenated `ParamSpec` in nested decorators due to a [mypy bug](https://github.com/python/mypy/issues/12909).

Reply:
Actually maybe it’s okay, since `validate_api_key` also sets `request.user`.

But wouldn’t this need a `request.user.is_authenticated` check for type narrowing?

Reply:
`is_staff` is also available on `AnonymousUser`, so type narrowing is not needed.